javascript-style-guide.md

```diff
@@ -34,24 +37,67 @@ avoid whitespace-oriented commits and diffs.
 
 Use JSCS to validate the format of your code. Your editor may offer JSCS
 integration, which you may find useful if you are not familiar with the style
-guide. You can also use our projects' Grunt interface.
+guide. You can also use our projects' Grunt or Gulp interfaces.
+
+# English
+
+## Spelling
 
-# Grammar
+Use real words for identifiers.
 
-## Use real words for identifiers.
+For instance, "element" is best written as "element", not "el", "elm", "elem",
+or "e".
 
-"element" is best written as "element", not "el", "elm", "elem", or "e".
+For your convenience, the following words also have a preferred spelling:
+
+Word      | Best written as | Known to be written as
+----------|-----------------|-----------------------
+directory | directory       | dir
+error     | error           | err, e
+index     | index           | idx, i
+request   | request         | req
+response  | response        | res
```

Comment:
While I agree with the general principle, I think we're being way too prescriptive here. We should spend our time making sure code is understandable than policing spelling and abbreviations. Consider this:

``` js
    var instance = getSideTracked(theCurrentStateOfAffairs);
    if (instance) {
      buyRosesForMom(instance);
    }
```
While containing no spelling errors, this chunk can mean anything. OTOH
``` js
    var err = tryToDoSomethingWith(thisObject);
    if (err) {
        handleError(err);
    }
```
Is quite clear, despite abbreviating the "error" variable.